src/docs/binding-tuples.md

```diff
@@ -0,0 +1,60 @@
+# Binding Tuples
+
+Tuples in Swift are conceptually anonymous structs. Apple implies that in the documentation by saying that tuples follow the same packing rules as structs when they are stored in memory. However, this is absolutely not the case when tuples are being passed as arguments. For whatever reason, when passed as arguments tuples are flattened entirely to individual elements that are not tuples. In other words, the following two functions are identical from an ABI perspective:
+```swift
+public func foo(T: (a: Int, b: Bool, (c: SomeStruct, d: SomeClass))) { // func of 1 argument
+    // ...
+}
+
+public func foo(a: Int, b: Bool, c: SomeStruct, d: SomeClass) { // func of 4 arguments
+    // ...
+}
+```
+It's unclear why Apple would do this except that it might be a vestige of when Swift had the ability to have multiple
+argument lists and could do partial function application. It also might be because it enables very low-friction usage of tuples in that when used outside of the context of nominal types, tuples don't actually need to exist. They can be implemented as an artifical grouping of scoped variable names.
+
+## Language Parity Mistmatch
+
+Since Tuples became available in C# they have had two varieties: `System.Tuple` and `System.ValueTuple`. The latter is what has been adopted into C# as a first class citizen in the language and since it matches closely to Swift tuples, we should follow that.
+
+One potential problem we need to investigate is that the C# `ValueTuple` is generic and the size of the struct can't be put into a `StructLayout` attribute. Furthermore, we need to know if we need to change `ValueTuple` to have `[StructLayout(LayoutKind.Sequential)]` added to the types. If not, it might be necessary to create our own type that can easily be converted back and forth to `ValueTuple`.
```

Comment:
For tuple fields in bitwise-copyable frozen structs, the Swift projection may need to define its own types, using the features in https://github.com/dotnet/runtime/issues/100896, to get correct layout rules (unless Swift separates out tuples fields like it does for parameters).

Reply:
Swift does not separate out tuple fields. Swift tuples get packed like structs using "Universal" layout. It's mentioned [here](https://github.com/swiftlang/swift/blob/main/docs/ABI/TypeLayout.rst), but this doc was last edited 6 years ago.

Reply:
In that case, the Swift projection may need to define its own tuple fields to get the correct layout experience. However, those types could have a `Deconstruct` method defined on them and an implicit conversion from/to ValueTuple to enable the C# features around value tuples.